File: golang/models/product.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Product struct {
	ID            uint           `gorm:"primaryKey" json:"id"`
	SupplierID    uint           `gorm:"not null;index" json:"supplier_id"`
	Supplier      User           `gorm:"foreignKey:SupplierID" json:"supplier,omitempty"`
	Name          string         `gorm:"not null" json:"name"`
	Description   string         `gorm:"type:text" json:"description"`
	SKU           string         `gorm:"uniqueIndex" json:"sku"`
	Price         float64        `gorm:"type:decimal(10,2);not null" json:"price"`
	StockQuantity int            `gorm:"default:0;not null" json:"stock_quantity"`
	Unit          string         `gorm:"type:varchar(20)" json:"unit"`
	Category      string         `gorm:"type:varchar(50)" json:"category"`
	ImageURL      string         `json:"image_url"`
	CreatedAt     time.Time      `json:"created_at"`
	UpdatedAt     time.Time      `json:"updated_at"`
	DeletedAt     gorm.DeletedAt `gorm:"index" json:"-"`
}

File: flutter/lib/models/product.dart
class Product {
  final int id;
  final int supplierId;
  final String name;
  final String? description;
  final String sku;
  final double price;
  final int stockQuantity;
  final String? unit;
  final String? category;
  final String? imageUrl;
  final DateTime createdAt;
  final DateTime updatedAt;
  final DateTime? deletedAt;

  Product({
    required this.id,
    required this.supplierId,
    required this.name,
    this.description,
    required this.sku,
    required this.price,
    required this.stockQuantity,
    this.unit,
    this.category,
    this.imageUrl,
    required this.createdAt,
    required this.updatedAt,
    this.deletedAt,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'] as int,
      supplierId: json['supplier_id'] as int,
      name: json['name'] as String,
      description: json['description'] as String?,
      sku: json['sku'] as String,
      price: (json['price'] as num).toDouble(),
      stockQuantity: json['stock_quantity'] as int,
      unit: json['unit'] as String?,
      category: json['category'] as String?,
      imageUrl: json['image_url'] as String?,
      createdAt: DateTime.parse(json['created_at'] as String),
      updatedAt: DateTime.parse(json['updated_at'] as String),
      deletedAt: json['deleted_at'] != null ? DateTime.parse(json['deleted_at'] as String) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'supplier_id': supplierId,
      'name': name,
      'description': description,
      'sku': sku,
      'price': price,
      'stock_quantity': stockQuantity,
      'unit': unit,
      'category': category,
      'image_url': imageUrl,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
      'deleted_at': deletedAt?.toIso8601String(),
    };
  }
}
